src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,154 +5,115 @@
-    public onDidChangeCodeLenses?: vscode.Event<void> | undefined
-
-    public async provideCodeLenses(
-        document: vscode.TextDocument,
-        token: vscode.CancellationToken
-    ): Promise<vscode.CodeLens[]> {
-        const search: TypescriptLambdaHandlerSearch = new TypescriptLambdaHandlerSearch(document.uri)
-        const handlers: LambdaHandlerCandidate[] = await search.findCandidateLambdaHandlers()
-
-        const lenses: vscode.CodeLens[] = []
-
-        handlers.forEach(handler => {
-            const range: vscode.Range = new vscode.Range(
-                document.positionAt(handler.positionStart),
-                document.positionAt(handler.positionEnd),
+import {  TypescriptLambdaHandlerSearch } from '../typescriptLambdaHandlerSearch'
+
+const getSamProjectDirPathForFile = async (filepath: string): Promise<string> => {
+    const packageJsonPath: string | undefined = await findFileInParentPaths(
+        path.dirname(filepath),
+        'package.json'
+    )
+    if (!packageJsonPath) {
+        throw new Error( // TODO: Do we want to localize errors? This might be confusing if we need to review logs.
+            localize(
+                'AWS.error.sam.local.package_json_not_found',
+                'Unable to find package.json related to {0}',
+                filepath
             )
-            const workspaceFolder:
-                vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(document.uri)
-
-            if (!workspaceFolder) {
-                throw new Error(`Source file ${document.uri} is external to the current workspace.`)
-            }
-
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, false, workspaceFolder))
-            lenses.push(this.generateLocalInvokeCodeLens(document, range, handler.handlerName, true, workspaceFolder))
+        )
+    }
 
-            try {
-                lenses.push(this.generateConfigureCodeLens(document, range, handler.handlerName, workspaceFolder))
-            } catch (err) {
-                const error = err as Error
+    return path.dirname(packageJsonPath)
+}
 
-                console.error(
-                    `Could not generate 'configure' code lens for handler '${handler.handlerName}': ${error.message}`
-                )
+export const initialize = ({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void => {
+    vscode.commands.registerCommand(
+        getInvokeCmdKey('javascript'),
+        async (args: LambdaLocalInvokeArguments) => {
+            let debugPort: number | undefined
+
+            if (args.debug) {
+                debugPort = 5858 // TODO: Use utility to find an available port
             }
-        })
 
-        return lenses
-    }
+            const activeFilePath = vscode.window.activeTextEditor!.document.uri.fsPath
```

Comment:
Why `activeTextEditor`/`activeFilePath` instead of `args.document.uri` ?

Reply:
That works too. Changed.
✔️